openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphHopper.java

```diff
@@ -230,8 +230,20 @@ else if (ALT_ROUTE.equalsIgnoreCase(algoStr))
 			EdgeFilter edgeFilter = edgeFilterFactory.createEdgeFilter(request.getAdditionalHints(), encoder, getGraphHopperStorage());
 			routingTemplate.setEdgeFilter(edgeFilter);
 
-			PathProcessor pathProcessor = pathProcessorFactory.createPathProcessor(request.getAdditionalHints(), encoder, getGraphHopperStorage());
-			ghRsp.addReturnObject(pathProcessor);
+
+			if (request.getAlgorithm().equals("alternative_route")) {
+				for (int c = 0; c < request.getHints().getInt("alternative_route.max_paths", 2); c++) {
```

Comment:
so this loop adds the same response object depending on the number of alternative routes requested? Is there a clearer way of writing this as at first glance it looks like there is something not quite right...

Reply:
I had the more concise version before and thought that would make it easier to read and understand, but ok...